### ISSUE TITLE
[Discussion] Private Genesis State

### ISSUE THREAD

**Issue by antonydenyer** [hyperledger/besu#2385]
### Description
As a user of unrestricted privacy, I want to be able to specify the initial genesis state for private transactions.

There are a couple of options on how to implement this:

1. separate private genesis config e.g `private-genesis.json`

2. add an option into genesis.config e.g 
```json
"alloc": {
  "0x0ffd23af8eebc60b3cfdeed6f814988757237314": {
    "private": true,
    "code": "0x6080...",
  }    
}
```

**Comment by antonydenyer** [comment]
There are also question about how much to implement as a first version. Perhaps a good start would be to use the same public genesis on the private state? This would cater for many scenarios.

**Comment by vmichalik** [comment]
I think there might be a pitfall to be avoided where transaction participants can have divergent private states

**Comment by antonydenyer** [comment]
> I think there might be a pitfall to be avoided where transaction participants can have divergent private states

Yeah, could be an issue but don't think there would be much gain in the world of private transactions. Another option would be to make it part of the `createPrivacyGroup` transaction (this would only work for onchain privacy groups) so that you have the initial members and the genesis as part of the first private transaction.

**Comment by antonydenyer** [comment]
The other option is to go down the plugin/factory approach where the user can apply whatever transactions they want to a group before it becomes 'active'. The tradeoff is that everything who's using privacy groups on that network must have the same plugin in installed. But, the same would be true for with config file.

**Comment by atoulme** [comment]
That sounds interesting. There is no spec for genesis files, I’d be very interested if we can land something here and port it to EEA tests.

**Comment by antonydenyer** [comment]
I didn't realise there wasn't anything in place! The first implementor wins I guess :smile:

**Comment by atoulme** [comment]
Agreed!

**Comment by vmichalik** [comment]
@antonydenyer I believe this discussion continued in another ticket? maybe I'm misremembering

**Comment by antonydenyer** [comment]
It did https://github.com/hyperledger/besu/pull/2509

I guess there are a couple of things to consider. 

1. what we do from a besu point of view
2. what should the standard look like

Personally, I think @pinges is right. Besu should implement it using a plugin architecture. Then from a standards point of view, we can provide an off the shelf plugin that implements/adheres to the standard.

**Comment by antonydenyer** [comment]
Discussed and merged in #2509
